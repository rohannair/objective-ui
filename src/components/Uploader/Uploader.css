/* @import '../../styles/globals';

.label {
  font-size: 1.2em;
  cursor: pointer;
  color: #ccc;
  padding: 5px;
  transition: color 0.25s ease-in-out;

  i {
    font-weight: bold;
  }

  &:hover {
    color: $primary;
  }

  &:active {
    position: relative;
    top: 1px;
  }
}

.fileInput {
  display: none;
}
 */
