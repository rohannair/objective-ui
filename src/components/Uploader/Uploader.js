import React, { Component, PropTypes } from 'react'
import styles from './Uploader.css'

import FormData from 'form-data'

class Uploader extends Component {
  static propTypes = {
    submitImage: PropTypes.func.isRequired,
    icon: PropTypes.string
  }

  render() {
    const { icon, submitImage } = this.props
    return (
      <label className={styles.label}>
        { icon && <i className={`zmdi zmdi-${icon}`} />}
        <input
          className={styles.fileInput}
          type="file"
          onChange={ this._validateUpload.bind(this, submitImage) }
          accept="image/*; capture=camera"
          ref={input => {
            this.fileInput = input
          }}
        />
      </label>
    )
  }

  _validateUpload = (cb, e) => {
    const uploadedFile = e.target.files[0]

    if (uploadedFile) {
      const reader  = new FileReader()
      reader.onload = function(ev) {
        let binaryString = ev.target.result
        cb(btoa(binaryString))
      }

      reader.readAsBinaryString(uploadedFile)
    }

    // cb(form)
  }
}

export default Uploader
