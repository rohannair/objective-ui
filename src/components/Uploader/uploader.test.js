import React from 'react'

import Uploader from './index.js'

describe('Test suite for Uploader component', () => {
  it('Uploader should exist', () => {
    let cb = jest.fn()
    let wrapper = shallow(<Uploader submitImage={cb} />)
    expect(wrapper).toMatchSnapshot()
  })
})

