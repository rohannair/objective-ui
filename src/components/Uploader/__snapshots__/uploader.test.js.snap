exports[`Test suite for Uploader component Uploader should exist 1`] = `
<label>
  <input
    onChange={[Function]}
    type="file" />
</label>
`;
